homeassistant/components/growatt_server/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Read status of growatt inverters."""
+import datetime
+import logging
+
+import growattServer
+
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_USERNAME, CONF_PASSWORD)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PLANT_ID = "plant_id"
+DEFAULT_NAME = "Growatt"
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PLANT_ID, default="0"): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Growatt sensor."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+    plant_id = config.get(CONF_PLANT_ID)
+    name = config.get(CONF_NAME)
+
+    api = growattServer.GrowattApi()
+
+    # Log in to api and fetch first plant if no plant id is defined.
+    login_response = api.login(username, password)
+    if not login_response["success"] and login_response["errCode"] == "102":
+        _LOGGER.error("Username or Password may be incorrect!")
+        return
+    user_id = login_response["userId"]
+    if plant_id == "0":
+        plant_info = api.plant_list(user_id)
+        plant_id = plant_info["data"][0]["plantId"]
+
+    # Get a list of inverters for specified plant to add sensors for.
+    inverters = api.inverter_list(plant_id)
+
+    entities = [GrowattInverter(api, "%s_Total" % name, plant_id, username, password)]
```

Comment:
Please use f-strings when string formatting is required.

Reply:
Sorry, i read the example wrong and picked the wrong method of formatting, it should be fixed now.